libraries/plugins/json_rpc/json_rpc_plugin.cpp

```diff
@@ -30,7 +33,7 @@ namespace detail
    {
       std::string                      jsonrpc = "2.0";
       fc::optional< fc::variant >      result;
-      fc::optional< json_rpc_error >   error;
+      json_rpc_error                   error;
```

Comment:
This field needs to be optional. By jsonrpc spec, the error field cannot exist unless there is an error.